Subject: 0005-upstream-DisableForwarding-yes-didn-t-override-Permi.patch: upstream: DisableForwarding=yes didn't override
CVE backport for Alpine 3.23 openssh 10.2_p1-r1 (aports 3.24-stable patches applied onto the tracked 10.2p1 tree).
Tracker: https://nvd.nist.gov/vuln/detail/CVE-2026-59999
Upstream source: https://github.com/openssh/openssh-portable/commit/8dfe7ed6e2fd988de08df508355a196b956b2753
Alpine aports patch: https://github.com/alpinelinux/aports/blob/da03ee144d27cece40d2a293c83214592415ae91/main/openssh/0005-upstream-DisableForwarding-yes-didn-t-override-Permi.patch
Base tree: OpenSSH portable 10.2p1 (https://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-10.2p1.tar.gz, sha256:ccc42c0419937959263fa1dbd16dafc18c56b984c03562d2937ce56a60f798b2), extracted as openssh-10.2p1/
Files: serverloop.c

Ported verbatim from Alpine aports 3.24-stable commit da03ee144d27cece40d2a293c83214592415ae91; the diff applies unchanged onto OpenSSH portable 10.2p1.

From 0e43a271759730085a507db9cbfc6d3e155b74d9 Mon Sep 17 00:00:00 2001
From: "djm@openbsd.org" <djm@openbsd.org>
Date: Sun, 31 May 2026 04:47:29 +0000
Subject: [PATCH 05/12] upstream: DisableForwarding=yes didn't override
 PermitTunnel=yes

Reported independently by Huzaifa Sidhpurwala of Redhat and Marko
Jevtic; ok markus@

OpenBSD-Commit-ID: b5c13f0746cf079b21f8deba47407fad49ccbf4c
(cherry picked from commit 8dfe7ed6e2fd988de08df508355a196b956b2753)
[CVE-2026-59999]
---
 serverloop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/serverloop.c b/serverloop.c
index 8e63480ecefa..42c3ce9fe60f 100644
--- a/serverloop.c
+++ b/serverloop.c
@@ -523,7 +523,7 @@ server_request_tun(struct ssh *ssh)
 		ssh_packet_send_debug(ssh, "Unsupported tunnel device mode.");
 		return NULL;
 	}
-	if ((options.permit_tun & mode) == 0) {
+	if ((options.permit_tun & mode) == 0 || options.disable_forwarding) {
 		ssh_packet_send_debug(ssh, "Server has rejected tunnel device "
 		    "forwarding");
 		return NULL;
-- 
2.55.0.dirty
